datastore/cache.go

```diff
@@ -71,6 +73,23 @@ func (cache *MemCache) Limit(n int) *MemCache {
 	return cache
 }
 
+func formKey(category, key string) string {
+	return category + "::" + key
+}
+
+// Get retrieves the value stored at the given key. If not set, returns a KeyNotFound error.
+func (cache *MemCache) Get(category, key string) (interface{}, error) {
+	key = formKey(category, key)
+	value, found := cache.instance.Get(key)
+	if found {
+		if err, ok := value.(error); ok {
+			return nil, err
+		}
+		return value, nil
+	}
+	return nil, KeyNotFound{key: key}
```

Comment:
Nit: Maybe a comment along the lines of "We return an error here since nil could also be a cached value" would be useful? I had to spend a second or two thinking about why this was here.

Reply:
Hmm, should we not do the same coersion of cached errors? We could return them as values and let the caller deal with them, but I think that would make this particular use more complex.

Reply:
Ah yeah we did cache errors. Ignore me then. I think what's here is OK.

Reply:
I think I simplified it a bit, give it another look.

Reply:
Yeah, I think what you have is OK. If people want to know whether a key exists in the cache, we could always add a `Exists` helper to the interface.

It sucks that we can't do something like `(value, err, ok)` due to Go-linting, but I guess our cache also caches errors so it's a bit different from a typical map (and that type-signature also reads weird).